src/device/router_device_worker.erl

```diff
@@ -1888,9 +1892,12 @@ channel_correction_and_fopts(Packet, Region, Device, Frame, Count, ADRAdjustment
                 %% `lorwan_mac_region'.  But `set_channels' wants data
                 %% rate in the form of "SFdd?BWddd?" so that's what
                 %% we'll give it.
-                NewDr = lorawan_mac_region:dr_to_datar(Region, NewDataRateIdx),
+                Plan = lora_plan:region_to_plan(Region),
+                NewDr = lora_plan:atom_to_datarate(
+                    (lora_plan:index_to_datarate(Plan, NewDataRateIdx))
+                ),
```

Comment:
This part seems a bit weird. Both are `_to_datarate` functions, but return different forms of datarate?
I'm not sure the best way to disambiguate "datarate" here.

Reply:
I've changed the API to these three APIs

datarate_to_index(Plan, DataRate)
datarate_to_atom(Plan, DataRate)
datarate_to_binary(Plan, DataRate)

The APIs will handle all three cases.  Definitely makes the code clearer.